fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanTransactionAssembler.java

```diff
@@ -65,4 +65,28 @@ LoanTransaction assembleTransactionAndCalculateChanges(Loan loan, JsonCommand co
         return LoanTransaction.repaymentType(repaymentTransactionType, loan.getOffice(), repaymentAmount, paymentDetail, transactionDate,
                 txnExternalId, null);
     }
+
+    LoanTransaction assembleAccrualActivityTransactionAndCalculateChanges(Loan loan, JsonCommand command, Map<String, Object> changes) {
```

Comment:
No needed this way. There will be no incoming command. The loan is needed and to calculate the balances.